src/java/org/apache/cassandra/service/ActiveRepairService.java

```diff
@@ -786,7 +781,7 @@ public synchronized ParentRepairSession removeParentRepairSession(TimeUUID paren
                 for (ColumnFamilyStore cfs : session.columnFamilyStores.values())
                 {
                     if (cfs.snapshotExists(snapshotName))
-                        cfs.clearSnapshot(snapshotName);
+                        StorageService.instance.snapshotManager.clearEphemeralSnapshot(snapshotName);
```

Comment:
If we hold ephemerals, we need to remove them from manager when repair wants to clear it.